### ISSUE TITLE
[Feature] Support for Huawei's HMS?

### ISSUE THREAD

**Issue by deakjahn** [kmcgill88/admob_flutter#223]
Considering that your approach with putting the ad into the widget tree is way nicer than the Firebase overlay, and also considering that a growing number of us developers are contemplating to support Huawei's parallel HMS ecosystem in order to make our users happier, did you ever think about adding support, maybe directly or as a parallel plugin? From as much as I can tell, the HMS API is more or less source compatible with Google's, with very small modifications needed about class names are referenced libraries.

**Comment by deakjahn** [comment]
I decided I need it badly enough to have a crack at it. :-) I cloned your plugin locally and modified what's needed to call the HMS equivalent routines. The example works with all three ad types now, just as expected, as far as I can tell. I'd be more than willing to send it to you for consideration. I couldn't find an e-mail on your profile page but there is one on mine. Maybe you could ping me?

**Comment by EliasDeuss** [comment]
@deakjahn I didn't know admob would work with HMS. Feel free to submit a pull request if you get it working. I don't have a Huawei device to test it out on though. I also sumitted a issue for dart-lang for HMS platform detection.

**Comment by deakjahn** [comment]
No, it's not a simple PR. It's a full, separate plugin that simply clones your code and modifies it to Huawei. It's not Admob, it's their own solution that works more or less just like Admob, the code is essentially the same, with very small differences, and with modified names everywhere. If you want to include it, I can send it the whole bundle but you might want to decide how to proceed with it.

Yes, I have a Huawei phone to test it with, and I have apps that use it successfully already in Huawei's store. The detection won't really help you, actually: Google simply refuses to accept any program with any HMS code in it, even if it's just detection, to the store (actually they said they would only do it from next January for new apps and next September for already published apps but they still do it already).
